### PR TITLE
[JENKINS-39738, JENKINS-43668] - Pick up SSHD module including sshd-core 1.6.0

### PR DESCRIPTION
**Pull request by jglick**
# Description

Integration of https://github.com/jenkinsci/sshd-module/pull/10 & https://github.com/jenkinsci/sshd-module/pull/17. Amends #2795, where @oleg-nenashev requested I omit the SSHD update to be on the safe side.

### Changelog entries

Proposed changelog entries:

* Major RFE: SSHD Module 2.0: Update from SSHD Core 0.14.0 to Apache MINA SSHD 1.6.0 in Jenkins core and Jenkins CLI
  * SSH Changelogs: https://github.com/jenkinsci/sshd-module/blob/master/CHANGELOG.md#20
  * Compatibility notice: https://github.com/jenkinsci/sshd-module/blob/master/CHANGELOG.md#20-compatibility-notice
* RFE: JENKINS-39738 - SSHD Module 2.0: Enable aes192ctr and aes256ctr ciphers if JVM supports unlimited-strength encryption
* Internal RFE: JENKINS-43668 - SSHD Module 2.0: Replace Trilead SSH-2 API usages by the built-in JVM functionality

### Desired reviewers

@reviewbybees & @jenkinsci/code-reviewers

### CODE REVIEW

**Comment by daniel-beck** [comment]
> `pluginFirstClassLoader`

A reminder that we still expose all core dependencies to plugins for no reason :cry:

**Comment by jglick** [comment]
@oleg-nenashev let me know if you want this amended to pick up https://github.com/jenkinsci/sshd-module/pull/14.

**Comment by oleg-nenashev** [comment]
@jglick Do you mean enabling Ciphers on the client side? It may be useful though I am not sure it is urgent to do so.

**Comment by oleg-nenashev** [comment]
On hold till we agree on the approach in #2848

**Comment by oleg-nenashev** [comment]
Started the thread in the MINA user list: http://www.mail-archive.com/users@mina.apache.org/msg06503.html , CC @aheritier

**Comment by jglick** [comment]
IIUC this is on hold pending SSHD 1.5.0 or some other resolution to a binary compatibility issue?

**Comment by oleg-nenashev** [comment]
@jglick Yes, the ball is on my side I'd say

**Comment by oleg-nenashev** [comment]
We are still blocked with this PR.

**Comment by paladox** [comment]
I wonder when 1.6.0 will be released. Since I presume this https://github.com/apache/mina-sshd/commit/b0c5ca22d413d67d5a857f3506eb840f58ae0a18 is the fix for the licensee they were talking about which was blocking 1.5.0.

**Comment by oleg-nenashev** [comment]
I ping them regularly

**Comment by paladox** [comment]
Ok thanks :)

**Comment by oleg-nenashev** [comment]
:(

**Comment by oleg-nenashev** [comment]
I could build a custom version to unblock us

**Comment by paladox** [comment]
@oleg-nenashev if you want and have time to do it :)

will this be for maven i.e. custom versioning it to use -jenkins prefix?

**Comment by oleg-nenashev** [comment]
@paladox Yes, it will be a custom jenkins-specific version in the Jenkins repo. I have offered my help to the SSHD maintainers multiple times, but seems there is no plausible ETA

File: cli/pom.xml
Comment (oleg-nenashev) [inline cli/pom.xml]: Should be 1.6.0

File: war/pom.xml
Comment (oleg-nenashev) [inline war/pom.xml]: 2.0 is out

**Review by oleg-nenashev — DISMISSED** [review]
SSHD 2.0 is out, we can proceed on this PR.
https://github.com/jenkinsci/sshd-module/blob/master/CHANGELOG.md#20

**Review by oleg-nenashev — APPROVED** [review]
🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Review by paladox — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I will update the changelogs tomorrow

**Comment by oleg-nenashev** [comment]
@paladox @jglick @daniel-beck Please take a look at the new changelog entries. Does it work for you? Surprisingly, I was unable to find a JIRA ticket perfectly matching the first Major RFE

**Comment by paladox** [comment]
@oleg-nenashev thanks, looks good to me :). Untested though, but there is not a big difference between 1.4.0 with 1.6.0 to cause big issues.

**Pull request merged**
